### DISCUSSION TITLE
Series of prompts

### DISCUSSION THREAD

**Discussion by ta5946** [open-webui/open-webui#6776] [Ideas]
Given some input variables, what would be the best way of executing a **predefined series of prompts** in the **background** and returning only the final model response? In my understanding these are called **promptbooks** and are used to divide a complex task into multiple prompts.
I read about pipelines but am checking if there is a simpler solution available.

For reference: https://learn.microsoft.com/en-us/copilot/security/using-promptbooks

**Comment by RPG-fan** [comment]
You could possibly chain an initial prompt to output to a predefined chat instance using a pipe with await response and inlet filter with chat parameters, and url parameters for the 2nd chat to call a specific model and a slice command to call a prompt, include an output filter to either continue calling models and prompts in the same chat, or forward to another instance, then have the last step return the response to the waiting pipe to output to the first chat. Seems plausible, but might need a bit of refinement.
